modules/tracking/samples/lasot_benchmark.py

```diff
@@ -0,0 +1,187 @@
+                elif tracker_name == "MOSSE":
+                    tracker = cv.TrackerMOSSE_create()
+                elif tracker_name == "CSRT":
+                    tracker = cv.TrackerCSRT_create()
+
+                init_once = False
+                init_bb = gt_bb
+                first_frame = cv.imread(image)
+
+            if not init_once:
+                temp = tracker.init(frame, init_bb)
+                init_once = True
+
+            #Update tracker state
+            temp, new_bb = tracker.update(frame)
+
+            #Evaluation of coordinates of corners and centers from [x, y, w, h]
+            new_bb_xmin = new_bb[0]
+            new_bb_xmax = new_bb[0] + new_bb[2] - 1.0
+            new_bb_ymin = new_bb[1]
+            new_bb_ymax = new_bb[1] + new_bb[3] - 1.0
+            gt_xmin = gt_bb[0]
+            gt_xmax = gt_bb[0] + gt_bb[2] - 1.0
+            gt_ymin = gt_bb[1]
+            gt_ymax = gt_bb[1] + gt_bb[3] - 1.0
+            new_cx = new_bb[0] + (new_bb[2] + 1.0) / 2
+            new_cy = new_bb[1] + (new_bb[3] + 1.0) / 2
+            gt_cx = gt_bb[0] + (gt_bb[2] + 1.0) / 2
+            gt_cy = gt_bb[1] + (gt_bb[3] + 1.0) / 2
+
+            # Width and height of overlap
+            dx = max(0, min(new_bb_xmax, gt_xmax) - max(new_bb_xmin, gt_xmin))
+            dy = max(0, min(new_bb_ymax, gt_ymax) - max(new_bb_ymin, gt_ymin))
+            area_of_overlap = dx * dy
+            area_of_union = (new_bb_xmax - new_bb_xmin) * (new_bb_ymax - new_bb_ymin) + (
+                gt_xmax - gt_xmin) * (gt_ymax - gt_ymin) - area_of_overlap
+            if (area_of_union != 0):
+                # Intersection over Union
+                intersection_over_union = area_of_overlap / area_of_union
+                sum_iou += intersection_over_union
+
+            precision = np.sqrt((new_cx - gt_cx) * (new_cx - gt_cx) + (new_cy - gt_cy) * (new_cy - gt_cy))
+            if precision > 20.0:
+                precision_value = 0.0
+            else:
+                precision_value = 1.0
+            sum_pr += precision_value
+
+            normalized_precision = np.sqrt((new_cx - gt_cx) / gt_bb[2] * (new_cx - gt_cx) / gt_bb[2] + (
+                new_cy - gt_cy) / gt_bb[3] * (new_cy - gt_cy) / gt_bb[3])
+            if normalized_precision < 0.5:
+                normalized_precision_value = 0.0
+            else:
+                normalized_precision_value = 1.0
+            sum_norm_pr += normalized_precision_value
+
+            gt_bb = gt_file.readline().replace("\n", "").split(",")
+
+        iou_values += sum_iou / frame_counter
+        pr_values += sum_pr / frame_counter
+        norm_pr_values += sum_norm_pr / frame_counter
+
+    mean_iou = iou_values / 280
+    mean_pr = pr_values / 280
+    mean_norm_pr = norm_pr_values / 280
+
+    results.write(tracker_name + ":\n\tmean IoU = " + str(mean_iou) + "\n\tmean precision = " + str(mean_pr) + "\n\tmean normalized precision = " + str(mean_norm_pr) + "\n\n")
```

Comment:
I think that it's better to print results on the screen

Reply:
OK.